### ISSUE TITLE
Upgrades duping using drawer removal

### ISSUE THREAD

**Issue by juh9870** [jaquadro/StorageDrawers#973]
If the drawer is removed in some way while upgrades screen is still open, you can still take out upgrades from the drawer that's already gone, then place the drawer back and repeat for infinite upgrades.
I tested it using Create mounted contraptions - when the contraption is assembled, the inventory screen remains open and you can take upgrades out

**Comment by juh9870** [comment]
this can be fixed by adding 

  ```java
  public boolean stillValid(PlayerEntity player) {
    if (tile.getLevel().getBlockEntity(tile.getBlockPos()) != tile ||
				player.distanceToSqr((double) tile.getBlockPos().getX() + 0.5D, (double) tile.getBlockPos().getY() + 0.5D, (double) tile.getBlockPos().getZ() + 0.5D) > 64.0D) return false;
    return true;
    }
```
to `com.jaquadro.minecraft.storagedrawers.inventory.InventoryUpgrade`

**Comment by jaquadro** [comment]
Thanks, this is fixed in the latest 1.16 release now
